### PR TITLE
cleanup meta.yaml and rerender

### PR DESCRIPTION
**Pull request by ickc**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy`
* [x] Ensured the license file is being packaged.

Supported Python versions are very old now: <https://anaconda.org/conda-forge/typogrify/files> Re-render to fix that.

### CODE REVIEW

**Comment by ickc** [comment]
@pmlandwehr

**Comment by isuruf** [comment]
If you add yourself as a maintainer, I'll merge.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,30 +1,24 @@
...
+  script: {{ PYTHON }} -m pip install . -vv
 
 requirements:
   host:
     - pip
     - python
-    - setuptools
-
   run:
     - python
```
Comment (isuruf) [inline recipe/meta.yaml]:
```suggestion
    - python >=3.6
```

**Comment by ickc** [comment]
> If you add yourself as a maintainer, I'll merge.

I have already listed myself as the 2nd maintainer.

**Comment by ickc** [comment]
Thanks, @isuruf!

**Pull request merged**